src/data/chats/chat.rs

```diff
@@ -398,15 +397,12 @@ impl Chat {
 
     pub fn send_message_to_agent(
         &mut self,
-        agent: MofaAgent,
+        agent: &MofaAgent,
         prompt: String,
-        mae_backend: &MofaBackend,
+        mofa_client: &MofaClient,
     ) {
         let (tx, rx) = mpsc::channel();
-        mae_backend
-            .command_sender
-            .send(SendTask(prompt.clone(), agent.clone(), tx.clone()))
-            .expect("failed to send message to agent");
+        mofa_client.send_message_to_agent(&agent, &prompt, tx);
 
```

Comment:
You may notice that in contrast to `send_message_to_model` and other similar operations, I'm not talking to a command sender directly but rather simply calling a method in the client.
I think having an indirection like this is clearer. The channel command handling is actually just a limitation of our thread and async handling (related to the client), not a meaningful part of the models or agents interface. I think previously we "leaked" those specifics into different parts of the UI and they just make the code harder to read and follow.
This will be slightly inconsistent for now but I think it's worth it, and either way we're looking into refactoring many aspects of this.